test/kes_dashboard_live_tiles_cli/TestOnlineLoader.h

```diff
@@ -12,24 +12,265 @@
+
+    OnlineLoader onlineLoader(
+        "/tmp/kes-dlt-cli/test", mockHttpClient, mockTileFactory);
+
+    // Ensuring the remainder of the function doesn't throw Exceptions by
+    // returning a valid KES response from the HTTPClient.GET().
+    ON_CALL(*mockHttpClient, GET_impl(OnlineLoader::KES_DLT_URL, _))
+        .WillByDefault(Return(this->response));
+
+    // Return the same mock Tile each for each tile data from the response.
+    ON_CALL(*mockTileFactory, create)
+        .WillByDefault(Return(mockTile));
+
+    // Ignore the call to Tile.initialise().
+    ON_CALL(*mockTile, initialise)
+        .WillByDefault(Return(true));
+
+    // Ignore the call to Tile.download().
+    ON_CALL(*mockTile, download)
+        .WillByDefault(Return(true));
+
+    EXPECT_CALL(*mockHttpClient, GET_impl(OnlineLoader::KES_DLT_URL, _))
+        .Times(1);
+
+    onlineLoader.getTiles();
+}
+
+/**
+ * Check that for a given HTTPClient.GET() response OnlineLoader.getTiles()
+ * calls for each tile data TileFactory.create().
+ */
+TEST_P(KesDltValidResponsesFixture, GetTilesCallsTileFactoryCreate) {
+    shared_ptr<MockHTTPClient> mockHttpClient = make_shared<MockHTTPClient>();
+    shared_ptr<MockTileFactory> mockTileFactory = make_shared<MockTileFactory>();  // NOLINT
+    shared_ptr<MockTile> mockTile = make_shared<MockTile>();
+
+    OnlineLoader onlineLoader(
+        "/tmp/kes-dlt-cli/test", mockHttpClient, mockTileFactory);
+
+    // Set the HTTPClient.GET to return a given valid KES response.
+    ON_CALL(*mockHttpClient, GET_impl(OnlineLoader::KES_DLT_URL, _))
+        .WillByDefault(Return(this->response));
+
+    // Return the same mock Tile each for each tile data from the response.
+    ON_CALL(*mockTileFactory, create)
+        .WillByDefault(Return(mockTile));
+
+    // Ignore the call to Tile.initialise().
+    ON_CALL(*mockTile, initialise)
+        .WillByDefault(Return(true));
+
+    // Ignore the call to Tile.download().
+    ON_CALL(*mockTile, download)
+        .WillByDefault(Return(true));
+
+    EXPECT_CALL(*mockTileFactory, create)
+        .Times(this->tileCount);
+
+    onlineLoader.getTiles();
+}
+
+
+MATCHER_P(AnyInJsonArray, jsonArray, "") {
+    for (int i = 0; i < json_array_get_count(jsonArray); i++)
+        if (json_value_equals(json_array_get_value(jsonArray, i), arg))
+            return true;
+    return false;
+}
+
+
+/**
+ * Check that for a given HTTPClient.GET() response OnlineLoader.getTiles()
+ * calls for each tile data Tile.initialise() with the expected JSON.
+ */
+TEST_P(KesDltValidResponsesFixture, GetTilesCallsTileInitialise) {
+    shared_ptr<MockHTTPClient> mockHttpClient = make_shared<MockHTTPClient>();
+    shared_ptr<MockTileFactory> mockTileFactory = make_shared<MockTileFactory>();  // NOLINT
+    shared_ptr<MockTile> mockTile = make_shared<MockTile>();
+
+    OnlineLoader onlineLoader(
+        "/tmp/kes-dlt-cli/test", mockHttpClient, mockTileFactory);
```

Comment:
Are you cleaning up this directory at the end? Also, maybe I've missed it but where is this directory being created?

Reply:
Cleaning up manually right now, need to sort it out - just haven't come up with a clean way of doing it yet 🤔 Directories right now are being created in Tile.download(). 

Actually, maybe ITileLoader should include something like `createCache`, `clearCache`.